.github/workflows/tests.yaml

```diff
@@ -0,0 +1,143 @@
+          echo "::set-output name=lint-flags::$(make lint-flags)"
+
+  lint:
+    needs:
+      - get-inputs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      # cache/restore go mod
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ needs.get-inputs.outputs.go-version }}
+
+      - name: go mod tidy
+        run: |
+          make mod
+          if ! git diff --exit-code; then
+            echo "Git directory has go.mod or go.sum changes"
+            exit 1
+          fi
+
+      - name: Lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: ${{ env.GOLANGCI_LINT_VERSION }}
+          args: ${{ needs.get-inputs.outputs.lint-flags }}
+          # Caching is covered by the actions/cache step above.
+          skip-cache: true
+
+
+  test:
+    needs:
+      - get-inputs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      # cache/restore go mod
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ needs.get-inputs.outputs.go-version }}
+
+      - name: Build
+        run: |
+          make e2e-container
+          docker save --output "${TARBALL_FILE}" e2e/vault-csi-provider:latest
+
+      - name: Test
+        run: make test
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: vault-csi-provider-image
+          path: ${{ env.TARBALL_FILE }}
+
+  integration-test:
+    needs:
+      - lint
+      - test
+    strategy:
+      fail-fast: false
+      matrix:
+        kind-k8s-version: [1.19.16, 1.20.15, 1.21.12, 1.22.9, 1.23.6, 1.24.2]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - run: npm install -g bats@${{ env.BATS_VERSION }}
+        shell: bash
+      - run: bats -v
+        shell: bash
+
+      - name: Create Kind Cluster
+        uses: helm/kind-action@v1.2.0
+        with:
+          cluster_name: kind
+          node_image: kindest/node:v${{ matrix.kind-k8s-version }}
+          version: ${{ env.KIND_VERSION }}
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: vault-csi-provider-image
+
+      - name: Load vault-csi-provider dev image
+        run: |
+          docker image load --input ${{ env.TARBALL_FILE }}
+          kind load docker-image e2e/vault-csi-provider:latest
```

Comment:
Not sure, but it looks like  the `e2e-setup` make target is already loading the docker image here: https://github.com/hashicorp/vault-csi-provider/blob/c4a6bdf92e947f348ff339b135e15f5d1d2b080b/Makefile#L66